pandas/parser.pyx

```diff
@@ -563,6 +563,18 @@ cdef class TextReader:
                 else:
                     raise ValueError('Python 2 cannot read bz2 from open file '
                                      'handle')
+            elif self.compression == 'zip':
+                import zipfile
```

Comment:
@jreback Thanks. I'll tackle this next along with pickles. I don't think this will be an overnight fix, but it's worth doing.